lib/msal-browser/src/controllers/UnknownOperatingContextController.ts

```diff
@@ -0,0 +1,530 @@
+
+        // Initialize performance client
+        this.performanceClient = this.config.telemetry.client;
+
+        // Initialize the crypto class.
+        this.browserCrypto = this.isBrowserEnvironment
+            ? new CryptoOps(this.logger, this.performanceClient)
+            : DEFAULT_CRYPTO_IMPLEMENTATION;
+
+        // Initialize the browser storage class.
+        this.browserStorage = this.isBrowserEnvironment
+            ? new BrowserCacheManager(
+                  this.config.auth.clientId,
+                  this.config.cache,
+                  this.browserCrypto,
+                  this.logger
+              )
+            : DEFAULT_BROWSER_CACHE_MANAGER(
+                  this.config.auth.clientId,
+                  this.logger
+              );
+
+        this.eventHandler = new EventHandler(this.logger, this.browserCrypto);
+    }
+    getBrowserStorage(): BrowserCacheManager {
+        return this.browserStorage;
+    }
+    getNativeInternalStorage(): BrowserCacheManager {
+        return this.browserStorage;
+    }
+    getNativeExtensionProvider(): NativeMessageHandler | undefined {
+        BrowserUtils.blockAPICallsBeforeInitialize(this.initialized);
+        BrowserUtils.blockNonBrowserEnvironment(this.isBrowserEnvironment);
+        return undefined;
+    }
+    setNativeExtensionProvider(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        provider: NativeMessageHandler | undefined
+    ): void {
+        BrowserUtils.blockAPICallsBeforeInitialize(this.initialized);
+        BrowserUtils.blockNonBrowserEnvironment(this.isBrowserEnvironment);
+        return undefined;
+    }
+    getNativeAccountId(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        request:
+            | PopupRequest
+            | RedirectRequest
+            | Partial<
+                  Omit<
+                      CommonAuthorizationUrlRequest,
+                      | "responseMode"
+                      | "codeChallenge"
+                      | "codeChallengeMethod"
+                      | "requestedClaimsHash"
+                      | "nativeBroker"
+                  >
+              >
+    ): string {
+        BrowserUtils.blockAPICallsBeforeInitialize(this.initialized);
+        BrowserUtils.blockNonBrowserEnvironment(this.isBrowserEnvironment);
+        return "";
+    }
+    getEventHandler(): EventHandler {
+        BrowserUtils.blockAPICallsBeforeInitialize(this.initialized);
+        BrowserUtils.blockNonBrowserEnvironment(this.isBrowserEnvironment);
+        return {} as EventHandler;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    getAccount(accountFilter: AccountFilter): AccountInfo | null {
+        BrowserUtils.blockAPICallsBeforeInitialize(this.initialized);
+        BrowserUtils.blockNonBrowserEnvironment(this.isBrowserEnvironment);
+        return null;
+    }
+
+    getAccountByHomeId(homeAccountId: string): AccountInfo | null {
```

Comment:
Not a fan of the code duplication this introduces. Any way to avoid reimplementing all of these?

Reply:
What are you thoughts on placing the implementation only in UnknownOperatingContextController?  This controller is always downloaded.  Another option would be to have a collection of functions that perform these operations.

Reply:
Now that you've moved to PublicClientNext where you expect initialize to be called before all of these, are these implementations still needed? Can we return null / empty array everywhere?

Reply:
This address the SSR rendering context.  If the window object is available.... initialize will just keep this controller.